### ISSUE TITLE
MacOS Catalina Cannot Save File

### ISSUE THREAD

**Issue by shhyou** [racket/gui#156]
On MacOS Catalina, I see a case where the user can modify existing files, but not saving new files.

## Steps to reproduce

1. Launch DrRacket.
2. (The language is Intermediate Student with Lambda; I forgot to try other languages.)
3. Click 'Save' or 'Save As'
4. Go to in `~/Downloads`, `~/` or `~/Comp Sci 111` and then click 'Save'

## Expected result

A new file is created in the selected folder.

## Actual result

Nothing saved. The tab name didn't change either.

## Racket versions tried

Racket v7.5 and Racket snapshot v7.6.0.3 e3d9090fd90457955e5949258b54fbe2f3a340af

## Other information

We tried giving DrRacket full disk access permission in System Preferences -> Security & Privacy -> Privacy but nothing happened.

Running `PLTSTDERR=warning drracket` from Terminal didn't print anything.

I don't know if this is related to racket/racket#2835 or not.

**Comment by rfindler** [comment]
I guess if you just run `put-file`, you see similar behavior?

**Comment by shhyou** [comment]
How to run `put-file`?

**Comment by shhyou** [comment]
We checked the directory using built-in Finder. We have not tried seeing if there's anything in the Open Recent menu, though and I cannot reproduce this on my own computer as well.

This is probably irrelevant: one difference I noticed was that Finder is configured to not show file extensions, and the save file dialogue did not show the `.rkt` extension either. Manually adding the file extension did not work though.
